### ISSUE TITLE
Research: Use storyshots to automate a11y testing

### ISSUE THREAD

**Issue by kat10140** [Esri/calcite-components#964]
### Background 

https://github.com/Esri/calcite-app-components/pull/963 enabled a tab to show a11y results that can be used to surface issues across use cases and different knob combinations. This relies on manually testing at the moment, but maybe https://github.com/storybookjs/storybook/tree/master/addons/storyshots/storyshots-puppeteer#axetest could be used to do automated a11y testing.

### Desired Outcome 

Use component stories and knob combinations to automatically run a11y testing.

https://github.com/Esri/calcite-app-components/issues/973

**Comment by jcfranco** [comment]
Closing in favor of https://github.com/Esri/calcite-components/issues/577

**Comment by jcfranco** [comment]
Actually, I'll reopen since this is related but not a duplicate. This will depend on #577's outcome.

**Comment by driskull** [comment]
@kat10140 can you update this issue to add a checkbox list for all components? Then we can divvy up the work.

**Comment by jcfranco** [comment]
@driskull @kat10140 That would be after. First we need to do the research/work to wire this up.

**Comment by julio8a** [comment]
Is this the accessibility tab on each component in the storybook? or is this something else?

**Comment by jcfranco** [comment]
It's somewhat related. Automating a11y audits with storybook stories. @eriklharper Could Screener be used for this?

**Comment by eriklharper** [comment]
Not sure how these a11y storybook features work, but as far as I know Screener only will work on exported functions from `.story` files that return JSX.  Either that or you use `screener-runner` manually to target html output of your choosing.  They support storybook but don't as far as I have inquired about support much in the way of any other storybook features other than the raw stories themselves.  I have asked about knob support and they currently don't have it and it is unclear if they plan on adding support for allowing devs to hook into storybook features like knobs to take screenshots based on knob interaction.  Loyal did mention to me that they plan to support Puppeteer, but to what level and extend I'm unsure of, and how that might apply to our particular setup.
